clients/src/test/resources/junit-platform.properties

```diff
@@ -13,3 +13,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 junit.jupiter.params.displayname.default = "{displayName}.{argumentsWithNames}"
+junit.jupiter.execution.timeout.default = 10 m
```

Comment:
Note that this sets a timeout of `10m` on the lifecycle methods as well. While I think we should have a timeout on lifecycle methods too, perhaps we can decrease its value to say `5m` (or even lesser if that seems apt)?  WDYT?

Refer the `junit.jupiter.execution.timeout.lifecycle.method.default` prop for more information.

Reply:
Also stumbled upon [Thread Mode](https://github.com/junit-team/junit5/pull/2949) which avoids pitfalls where the test code [doesn't interrupt](https://github.com/junit-team/junit5/issues/2087).